src/Compilers/CSharp/Portable/Compilation/SyntaxTreeSemanticModel.cs

```diff
@@ -2226,7 +2226,15 @@ private void ValidateStatementRange(StatementSyntax firstStatement, StatementSyn
                 throw new ArgumentException("statements not within tree");
             }
 
-            if (firstStatement.Parent == null || firstStatement.Parent != lastStatement.Parent)
+            // Global statements don't have their parent in common, but should belong to the same compilation unit
```

Comment:
I'm a bit confused by this comment. Aren't global statements having compilation units as their parent? If the belong to the same compilation unit, then they have a common parent. Is there a case where global statement have a parent other than compilation unit?

Reply:
Yeah, the structure here is a little confusing. A top level statement has a `GlobalStatementSyntax` as its parent, which has the `CompilationUnitSyntax` as its parent. So two global statements in a file don't have a common parent, only a common grandparent. The `GlobalStatementSyntax` nodes themselves can't be passed to this method because they don't inherit from `StatementSyntax`.